Source/Core/DolphinQt/MenuBar.cpp

```diff
@@ -1218,6 +1224,74 @@ void MenuBar::GenerateSymbolsFromRSO()
   }
 }
 
+void MenuBar::GenerateSymbolsFromRSOAuto()
+{
+  const std::array<std::string, 2> search_for = {".elf", ".plf"};
+  std::vector<u32> matches;
+  AddressSpace::Accessors* accessors = AddressSpace::GetAccessors(AddressSpace::Type::Effective);
+
+  // Find filepath to elf/plf commonly used by RSO modules
+  for (const auto& str : search_for)
+  {
+    u32 next = 0;
+    while (true)
+    {
+      auto found_addr =
+          accessors->Search(next, (const u8*)str.data(), static_cast<u32>(str.size() + 1), true);
+
+      if (!found_addr.has_value())
+        break;
+      next = *found_addr + 1;
+
+      // Get the beginning of the string
+      found_addr = accessors->Search(*found_addr, (const u8*)"", 1, false);
+      if (!found_addr.has_value())
+        continue;
+
+      // Get the string reference
+      u32 ref_addr = *found_addr + 1;
+      const std::array<u8, 4> ref = {(ref_addr >> 24) & 0xFF, (ref_addr >> 16) & 0xFF,
```

Comment:
These should likely contain a cast around the values, rather than a mask, as this is implicitly truncating `u32` values down to `u8` as far as the type system cares.